build/deploy/main.go

```diff
@@ -0,0 +1,105 @@
+// main handles deployment of the plugin to a development server using either the Client4 API
+// or by copying the plugin bundle into a sibling mattermost-server/plugin directory.
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mholt/archiver/v3"
+	"github.com/pkg/errors"
+)
+
+func main() {
+	err := deploy()
+	if err != nil {
+		log.Print(err.Error())
+		os.Exit(1)
+	}
+}
+
+// deploy handles deployment of the plugin to a development server.
+func deploy() error {
+	if len(os.Args) < 3 {
+		return errors.New("deploy <plugin id> <bundle path>")
+	}
+
+	pluginId := os.Args[1]
```

Comment:
Given that this is freshly written code and we are encouraging plugin authors to write golint compliant code, I would like to see us following the convention here and writing golint compliant code our self.